File: HARVARD/cs50x/Soluções_CS50_Harvard/Linguagem_Python/week6/readability/readability.py
# Matheus Henrique dos Santos, https://github.com/mtheusdev  UDESC CCT CS50X - HARVARD
from cs50 import get_string


entrada = get_string("entrada: \n")

letras = 0
palavras = 0
sentences = 0
contador = 0

for i in entrada:
    contador += 1

for i in range(contador):
    if (ord(entrada[i]) >= 65 and ord(entrada[i]) <= 122):
        letras += 1

    elif (ord(entrada[i]) == 32 and (ord(entrada[i - 1]) != 33 and ord(entrada[i - 1]) != 46 and ord(entrada[i - 1]) != 63)):
        palavras += 1
    elif (ord(entrada[i]) == 33 or ord(entrada[i]) == 46 or ord(entrada[i]) == 63):
        sentences += 1
        palavras += 1

L = letras * 100 / palavras
S = sentences * 100 / palavras

index = round(0.0588 * L - 0.296 * S - 15.8)

if (index < 1):
    print("Before Grade 1")

elif (index >= 16):
    print("Grade 16+")

else:
    print(f"Grade {index}")

File: HARVARD/cs50x/Soluções_CS50_Harvard/Linguagem_C/week2/readability/readability.c
/* Matheus Henrique dos Santos, https://github.com/mtheusdev  UDESC CCT CS50X - HARVARD*/
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

//Implemente um programa que calcule o nível aproximado necessário para compreender algum entradao
//a formula padrao utilizada é index = 0.0588 * L - 0.296 * S - 15.8
int main(void)
{

    string entrada = get_string("entrada: \n");

    float letras = 0;
    float palavras = 0;
    float sentence = 0;


    for (int i = 0, n = strlen(entrada); i < n; i++)
    {

        if (entrada[i] >= 65 && entrada[i] <= 122)
        {
            letras++;
        }

        else if (entrada[i] == 32 && (entrada[i - 1] != 33 && entrada[i - 1] != 46 && entrada[i - 1] != 63))
        {
            palavras++;
        }

        else if (entrada[i] == 33 || entrada[i] == 46 || entrada[i] == 63)
        {
            sentence++;
            palavras++;
        }
    }

    float L = letras * 100 / palavras;
    float S = sentence * 100 / palavras;
    float index = round(0.0588 * L - 0.296 * S - 15.8);

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int) index);
    }
}
